archiver.rb

```diff
@@ -0,0 +1,40 @@
+class Archiver
+  attr_accessor :file
+  @@modes = %i[reader writer both]
+
+  @@modes.each do |mode|
+    define_method("open_file_as_#{mode}") do |file|
+      @file = File.open(file, "r") if mode == :reader
```

Comment:
Prefer use` ""` for interpolations. If you use just string `''`

Reply:
Correct, I will change this.